staging/src/k8s.io/legacy-cloud-providers/azure/azure.go

```diff
@@ -205,6 +205,9 @@ type Config struct {
 	VmssCacheTTLInSeconds int `json:"vmssCacheTTLInSeconds,omitempty" yaml:"vmssCacheTTLInSeconds,omitempty"`
 	// VmssVirtualMachinesCacheTTLInSeconds sets the cache TTL for vmssVirtualMachines
 	VmssVirtualMachinesCacheTTLInSeconds int `json:"vmssVirtualMachinesCacheTTLInSeconds,omitempty" yaml:"vmssVirtualMachinesCacheTTLInSeconds,omitempty"`
+	// VmssDeletedVirtualMachinesCacheTTLInSeconds sets the cache TTL for deleted vmssVirtualMachines.
+	// It is used to avoid throttling when large number of VirtualMachins are deleted all together and its value should be bigger than vmssVirtualMachinesCacheTTLInSeconds.
+	VmssDeletedVirtualMachinesCacheTTLInSeconds int `json:"vmssDeletedVirtualMachinesCacheTTLInSeconds,omitempty" yaml:"vmssDeletedVirtualMachinesCacheTTLInSeconds,omitempty"`
```

Comment:
> The cache above is too long and should be reduced to a shorter value when required

So when would we set this `VmssDeletedVirtualMachinesCacheTTLInSeconds` value and what's the suggested value in `Azure Node is recreated in a short time`?

Reply:
it depends on how long customers expect to recreate Nodes with the same Node name. And setting it to -1 would disable caches for deleted VMs.